paz/processors/keypoints.py

```diff
@@ -218,3 +223,122 @@ def __init__(self, parents):
+
+
+class StandardizeKeypoints2D(Processor):
+    def __init__(self, data_mean2D, data_stdev2D):
+        """ Processor class for standerize
+
+        # Arguments
+            data_mean2D: mean 2D
+            data_stdev2D: standard deviation 2D
+
+        # Return
+            standerized keypoints2D
+        """
+        self.mean = data_mean2D
+        self.stdev = data_stdev2D
+        super(StandardizeKeypoints2D, self).__init__()
+
+    def call(self, keypoints2D):
+        return standardize(keypoints2D, self.mean, self.stdev)
+
+
+class UnnormalizeData(Processor):
+    def __init__(self, data_mean3D, data_stdev3D, dim_to_use):
+        """ Processor class for unnormalize function in human36m.py
+
+        # Arguments
+            data_mean3D: mean 3D
+            data_stdev3D: standard deviation 3D
+            dim_to_use: dimensions to use
+
+        # Return
+            Unormalized data
+        """
+        self.mean = data_mean3D
+        self.stdev = data_stdev3D
+        self.valid = dim_to_use
+        super(UnnormalizeData, self).__init__()
+
+    def call(self, keypoints2D):
+        data = keypoints2D.reshape(-1, 48)
+        rearanged_data = np.zeros((len(data), len(self.mean)),
+                                  dtype=np.float32)
+        rearanged_data[:, self.valid] = data
+        unnormalized_data = destandardize(rearanged_data, self.mean,
+                                          self.stdev)
+        return unnormalized_data
+
+
+class SimpleBaselines3D(Processor):
+    def __init__(self, model, data_mean2D, data_stdev2D, data_mean3D,
+                 data_stdev3D, dim_to_use3D, args_to_mean,
+                 h36m_to_coco_joints2D):
+        """ Processor class to predict the 3D keypoints
+
+        # Arguments
+            model: Simplebaseline 3D model
+            data_mean2D: mean 2D
+            data_stdev2D: standard deviation 2D
+            data_mean3D: data mean 3D
+            data_stdev3D: standar deviation 3D
+            dim_to_use: dimensions to use
+            args_to_mean: joints indices
+            h36m_to_coco_joints2D: h36m data joints indices
+
+        # Return
+            keypoints2D: 2D keypoints
+            keypoints3D: 3D keypoints
+        """
+        super(SimpleBaselines3D, self).__init__()
+        self.model = model
+        self.merge_keypoints2D = MergeKeypoints2D(args_to_mean)
+        self.filter = FilterKeypoints2D(args_to_mean, h36m_to_coco_joints2D)
+        self.preprocess = StandardizeKeypoints2D(data_mean2D, data_stdev2D)
+        self.postprocess = UnnormalizeData(data_mean3D, data_stdev3D,
+                                           dim_to_use3D)
+
+    def call(self, keypoints2D):
+        keypoints2D = self.merge_keypoints2D(keypoints2D)
+        keypoints2D = self.filter(keypoints2D)
+        keypoints3D = predict(keypoints2D, self.model, self.preprocess,
+                              self.postprocess)
+        return keypoints2D, keypoints3D
```

Comment:
I would try to make a sequential processor for the preprocessing and have in the constructor a pr.Predict processor using it.

```
preprocess = pr.Sequential([MergeKeypoints2D(), FilterKeypoints2D(), StandardizeKeypoints2D()]
self.predict = pr.Predict(model, preprocess, postprocess)
```
and inside the call

```
def call(self, keypoints2D):
    keypoionts3D = self.predict(keypotsin2D)
    return keypoints2D, keypoints3D
```